packages-ts/starknet-gauntlet/src/commands/base/executeCommand.ts

```diff
@@ -205,6 +227,29 @@ export const makeExecuteCommand = <UI, CI>(config: ExecuteCommandConfig<UI, CI>)
       return tx
     }
 
+    declareContract = async (): Promise<TransactionResponse> => {
+      deps.logger.info(`Declaring contract ${config.category}`)
+      await deps.prompt('Continue?')
+      deps.logger.loading(`Sending transaction...`)
+
+      const tx = await this.provider.declareContract(
+        this.contract,
+        this.compiledContractHash,
+        false,
+      )
+
+      deps.logger.loading(`Waiting for tx confirmation at ${tx.hash}...`)
+      const response = await tx.wait()
+
+      if (!response.success) {
```

Comment:
I'm not sure this `success` flag is set properly. Please double check the [wait()](https://github.com/smartcontractkit/chainlink-starknet/blob/bf537fb7faf93eb4f516fcc47990b4d527e6c99c/packages-ts/starknet-gauntlet/src/provider/index.ts#L54) method.

It expects the underlying `provider.waitForTransaction` to throw, and if it doesn't throw it sets the `success` flag. The problem is it looks like the underlying fn will return a status enum, not throw.

Reply:
I can confirm it's set as expected -- we use this pattern already elsewhere in the code and looking at https://github.com/0xs34n/starknet.js/blob/b2e7a4046c8bda541cf0228469a318443d94e788/src/provider/rpc.ts#L405 it does seem that error states trigger an exception to be thrown